arch/arm/core/mpu/CMakeLists.txt

```
@@ -5,6 +5,7 @@ zephyr_library()
 zephyr_library_sources(                             arm_core_mpu.c)
 zephyr_library_sources_ifdef(CONFIG_CPU_HAS_ARM_MPU arm_mpu.c)
 zephyr_library_sources_ifdef(CONFIG_CPU_HAS_NXP_SYSMPU nxp_mpu.c)
+zephyr_library_sources_ifdef(CONFIG_CPU_CORTEX_M cortex_m/mpu.c)
```

Nit: we could change this to
```suggestion
if (CONFIG_CPU_CORTEX_M AND CONFIG_CPU_HAS_ARM_MPU)
    zephyr_library_sources(cortex_m/mpu.c)
endif()
```
to avoid needing to `#ifdef` the entire file